docs/source/tutorial/train_loop.rst

```diff
@@ -0,0 +1,549 @@
+How to write a training loop in Chainer
+```````````````````````````````````````
+
+.. currentmodule:: chainer
+
+In this tutorial section we will learn how to train a deep neural network to
+classify hand-written digits using the popular MNIST dataset. This dataset
+contains 50000 training examples and 10000 test examples. Each example contains
+a 28x28 greyscale image and a corresponding class label for the digit. Since
+the digits 0-9 are used, there are 10 class labels.
+
+Chainer provides a feature called :class:`~chainer.training.Trainer` that can
+be used to simplify the training process. However, we think it is good for
+first-time users to understand how the training process works before using the
+:class:`~chainer.training.Trainer` feature. Even advanced users might sometimes
+want to write their own training loop and so we will explain how to do so here.
+
+The complete training process consists of the following steps:
+
+1. Prepare a datasets that contain the train/validation/test examples.
+2. Optionally set of iterators for the datasets.
+3. Write a training loop that performs the following operations in each iteration:
+    A. Retreive batches of examples from the training dataset.
+    B. Feed the batches into the model.
+    C. Run the forward pass on the model to compute the loss.
+    D. Run the backward pass on the model to compute the gradients.
+    E. Run the optimizer on the model to update the parameters.
+    F. (Optional): Ocassionally check the performance on a validation/test set.
+
+1. Prepare the dataset
+''''''''''''''''''''''
+
+Chainer contains some built-in functions that can be used to download and
+return Chainer-formatted versions of popular datasets used by the ML and deep
+learning communities. In this example, we will use the built-in function that
+retrieves the MNIST dataset.
+
+.. testcode::
+
+    import matplotlib.pyplot as plt
+    from chainer.datasets import mnist
+
+    # Download the MNIST data if you haven't downloaded it yet
+    train, test = mnist.get_mnist(withlabel=True, ndim=1)
+
+    # Display an example from the MNIST dataset.
+    # `x` contains the input image array and `t` contains that target class
+    # label as an integer.
+    x, t = train[0]
+    plt.imshow(x.reshape(28, 28), cmap='gray')
+    plt.show()
+    print('label:', t)
+
+Output
+......
+
+.. image:: ../../image/train_loop/5.png
+
+label: 5
+
+2. Create the dataset iterators
+''''''''''''''''''''''''''''''
+
+Although this is an optional step, it can often be convenient to use iterators
+that operate on a dataset and return a certain number of examples (often called
+a "mini-batch") at a time. The number of examples that is returned at a time is
+called the "batch size" or "mini-batch size." Chainer already has an Iterator
+class and some subclasses that can be used for this purpose and it is
+straightforward for users to write their own as well.
+
+We will use the :class:`~chainer.iterators.SerialIterator` subclass of
```

Comment:
- "the" is not necessary?
- How about: We will use ..., which is a subclass of ...?